mapusaurus/hmda/views.py

```diff
@@ -16,32 +16,39 @@ def loan_originations(request):
     lender_hierarchy = request.GET.get('lh')
     peers = request.GET.get('peers')
     geoids = get_censustract_geoids(request)
-    
     institution_selected = Institution.objects.get(pk=institution_id)
     metro_selected = Geo.objects.filter(geo_type=Geo.METRO_TYPE, geoid=metro).first()
-    action_taken_selected = action_taken_param.split(',')
+    if action_taken_param:
+        action_taken_selected = [param for param in action_taken_param.split(',')]
+    else:
+        action_taken_selected = []
     if geoids and action_taken_selected:
         query = HMDARecord.objects.filter(
                 property_type__in=[1,2], owner_occupancy=1, lien_status=1,
-                action_taken__in=action_taken_selected)
+                action_taken__in=action_taken_selected,
+                geo_id__in=geoids)
         if lender_hierarchy == 'true':
             hierarchy_list = LenderHierarchy.objects.filter(organization_id=institution_selected.lenderhierarchy_set.get().organization_id)
             if len(hierarchy_list) > 0:
-                query = query.filter(institution__in=hierarchy_list) 
+                query = query.filter(institution__in=[item.institution for item in hierarchy_list]) 
```

Comment:
Good catch but is it really necessary? Does Django ORM handle the old way? I don't know, but its worth a look!